crates/accelerate/src/unitary_synthesis.rs

```diff
@@ -268,7 +268,7 @@ fn py_run_main_loop(
                     new_ids,
                     min_qubits,
                     target,
-                    coupling_edges,
+                    coupling_edges.clone(),
```

Comment:
We can avoid this clone if we refactor the functions a bit to split the python interface out and use a reference for the recursion. Not sure how important this is because this only comes up during control flow processing and we're already slower there because of the python usage and circuit to dag conversion. Maybe something to fix for when we make a dag native control flow op.

Reply:
I think it would make sense to revisit this when we get to have rust-native control flow op.